### DISCUSSION TITLE
We don't have good wsyiwyg markdown editors today: zim can be 'the one'

### DISCUSSION THREAD

**Discussion by quesada** [zim-desktop-wiki/zim-desktop-wiki#1906] [Ideas]
We don't have good wsyiwyg markdown editors today (said someone on the discussions here, and I agree).

There's a reason for that. Most attempts are built with electron, which is a sure recipe to have an app that 'sucks' (terrible typing latency, memory usage off the roof, etc).

The closest thing is typora, and it's very far in features from zim. Not really a notetaker, mostly designed for single doc use. Codemirror 6 seems to be getting there but still quite glitchy. This is what obsidian is using now to offer live preview, what they call their wsyiwyg markdown editor.

Markdown is in @jaap-karssenberg mind mid term. Here's why I think there's hope for this to move to 'short term':

About markdown support: I'm using now @pfalcon fork with it (https://github.com/pfalcon/zim-desktop-wiki-markdown) ! it seems to be usable, he uses it for day to day duty. There are some gitches but it fees mature enough that it could be a great candidate to be included into the main repo (@jaap-karssenberg , this is of course your decision; I'd be happy to do some exhaustive testing). It's not only me being happpy with the fork's implemetation:

![image](https://user-images.githubusercontent.com/182207/152636105-9822f01a-f992-41a7-aeae-4fe782a48655.png)

The fork author, @pfalcon 👍🏻 
![image](https://user-images.githubusercontent.com/182207/152636537-97fd2801-748b-478f-b13f-5b79078a7321.png)

There are some minor issues:
![image](https://user-images.githubusercontent.com/182207/152635991-33bd7e44-cbc7-4fa5-b106-cd343935c02b.png)
And one of them has a workaround: just don't have 'empty files' in the notebook: https://github.com/pfalcon/zim-desktop-wiki-markdown/issues/4

Having established that this md backend 'mostly works', I'm going to try to convince you that it's worth doing :)

1. Markdown allows integration with other systems. 
	a. copy pasting to forums, wikis, github, stackoverflow
	b. using 3rd party apps that can take .md as input but not .zimwiki. For example I use obsidian to visualize the graph of notes in a dynamic way. Couldn't do this with the zimwiki syntax before
	c. You can collaborate with others synchronously using things lilke hackMD (like gdocs, two cursors editing) easier roundtrip to zim
2. There are no good wysiwyg markdown editors, and the market for this is gigantic
3. As soon as all those users of electron apps experience how fast zim does things, it's likely they will start using it ☺ And they come from native markdown (typora, obsidian, logseq) OR can only export their notes to markdown and may be too lazy to run pandoc to get to zimwiki format. We should make their life easier :)

**Comment by introt** [comment]
While I do appreciate your enthusiasm @quesada, I'm afraid you're just creating more noise here.

To quote [Jaap on markdown support](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/26#issuecomment-514361355):

> **All**: there is no need to add comments just to ourselves how important this is. The only thing that will pull this in is actual code being submitted. If it is up to me to fix it please be very patient. Do realize that I'm doing this on my own outside of my actual working hours and as I'm already fulltime employed there is no way I can spent more time on zim.
>
> So please submit your own code - stable code with a test suite - or be patient.

Jaap's currently doing related refactoring (see my [related comment](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1879#issuecomment-1019532070) with relevant links) so progress is being made in the present.

As for the fork, the author is "not really aiming for anything, except for having a tool to maintain [their] Markdown-based personal knowledgebase" (https://github.com/pfalcon/zim-desktop-wiki-markdown/issues/2#issuecomment-896827293). While the fork is usable for end users, its implementation details does not align with this project's goals, making it hard to upstream their efforts.

Regarding Zim's competitiveness, some features being primitive or slow (tables, #1900, ...), etc. – Zim is not a product. We're just a bunch of volunteers doing our own thing at our own pace, and that's okay; not everything has to be about marketability or user counts. Besides, being free software, Zim's code is up for grabs for anyone who wants to take a different path.
